plugins/nf-validation/src/main/nextflow/validation/ImmutableMap.groovy

```diff
@@ -0,0 +1,214 @@
+package nextflow.validation
+
+// A class that works like Map, but returns an immutable copy with each method
+class ImmutableMap implements Map {
```

Comment:
Perhaps `ImmutableMap` should inherit from a concrete Map class rather than implementing the `Map` interface. This would ensure that we get the normal `meta["key"]` access as [noted by Júlia](https://github.com/nextflow-io/nf-validation/pull/32#issuecomment-1562602880).

```suggestion
class ImmutableMap extends LinkedHashMap {
```

Reply:
Thank you for the suggestion!